receipt_layoutlm/receipt_layoutlm/trainer.py

```
@@ -34,6 +34,33 @@ def _checkpoint_step(path: str) -> int:
+    Writing to /tmp/receipt_layoutlm/{job_name}/ defeats that
+    mechanism: the spot-restart pulls an empty /opt/ml/checkpoints/
+    back from S3, the trainer's auto-resume looks in /tmp/ and finds
+    nothing, and training silently starts over from epoch 0.
+
+    Resolution order:
+      1. /opt/ml/checkpoints (when SageMaker mounted it)
+      2. /tmp/receipt_layoutlm/{job_name} (local dev fallback)
+    """
+    if os.path.isdir(SAGEMAKER_CHECKPOINT_DIR):
+        return SAGEMAKER_CHECKPOINT_DIR
+    return f"/tmp/receipt_layoutlm/{job_name}"
```

Ruff (0.15.13)

[error] Probable insecure usage of temporary file or directory: "/tmp/receipt_layoutlm/"

(S108)